imslp.js

```diff
@@ -0,0 +1,239 @@
+			var movements = rows[i].cells[1].innerHTML;
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Year/Date of Composition")) {
+			var date = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("First Performance")) {
+			var place = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("First Publication")) {
+			var publisher = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Language") {
+			var language = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim().includes("Average Duration")) {
+			var duration = rows[i].cells[1].innerText.trim();
+		}
+		if (rows[i].cells[0].innerText.trim() == "Piece Style") {
+			var style = rows[i].cells[1].innerText.trim();
+		}
+
+		/* The metadata of instrumention is too mess.
+		if (rows[i].cells[0].innerText.trim() == "Instrumentation") {
+			var instrumentation = rows[i].cells[1].innerText.trim();
+		}*/
+		//Z.debug(instrumentation);
+	}
+
+	//push metadata
+	item.title = title.replace(/\(.*\)/, "");
+	item.url = url;
+	item.libraryCatalog = null;
+	if (author) {
+		item.creators.push(ZU.cleanAuthor(author, "author", true));
+		item.tags.push(author.split(',')[0]);
+	}
+	if (names) {
+		item.abstractNote = names.replace(/\[.*\]/, "");
+	}
+	if (movements) {
+		var movementsNotes = movements;
+		item.notes.push({ note: movementsNotes });
+	}
+	if (seriesNumber) {
+		item.seriesNumber = seriesNumber;
+	}
+	if (place) {
+		item.place = "First Performance: " + place;
+	}
+	if (publisher) {
+		item.publisher = publisher;
+	}
+	if (date) {
+		item.date = date;
+	}
+	if (language) {
+		item.language = language;
+	}
+	if (duration) {
+		item.extra = "Type: musical_score\nDuration: " + duration;
+	}
+	else {
+		item.extra = "Type: musical_score";
+	}
+	if (key) {
+		item.tags.push(key);
+	}
+	if (incipit) {
+		var incipitNotes = "Text Incipit:" + incipit.replace(/<i(([\s\S])*?)<\/i>/, "");
+		item.notes.push({ note: incipitNotes });
+	}
+	if (style) {
+		item.tags.push(style);
+	}
+
+	/* see 122
+	if (instrumentation) {
+		var instrumentationNotes = "Instrumentation\n" + instrumentation;
+		item.notes.push({ note: instrumentationNotes });
+	}*/
+
+	item.complete();
+}
+
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://imslp.org/wiki/Das_Lied_von_der_Erde_(Mahler%2C_Gustav)",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "Das Lied von der Erde",
+				"creators": [
+					{
+						"firstName": "Gustav",
+						"lastName": "Mahler",
+						"creatorType": "author"
+					}
+				],
+				"date": "1908–09",
+				"abstractNote": "大地之歌; Pieseň o zemi; Песнь о земле; השיר על הארץ; 대지의 노래;",
+				"extra": "Type: musical_score",
+				"language": "German",
+				"place": "First Performance: 1911-11-20 in Munich, Tonhalle.\nSara Cahier (alto), William Miller (tenor), Orchestra, Bruno Walter (conductor)",
```

Comment:
That's performance information -- we'd want publication info (i.e. Vienna, in this case)

Reply:
Just removed this until a suitable filed